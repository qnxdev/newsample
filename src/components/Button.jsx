function Button({ fn, children, title }) {
  return (
    <button onClick={fn} title={title} className="Button">
      {children}
    </button>
  );
}

export default Button;
