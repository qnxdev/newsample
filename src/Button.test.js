import React from "react";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "./components/Button";

describe("Button", () => {
  it("should render without issues", () => {
    const component = shallow(<Button />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should call onClick event', () => {
    const result = 'I was Pressed';
    const mockFn = jest.fn(() => result);
    const component = shallow(<Button fn={mockFn} />);

    expect(mockFn).not.toHaveBeenCalled();

    component.props().onClick();

    expect(mockFn.mock.calls.length).toBe(1);
    expect(component.props().onClick()).toBe(result);
  });
  it('button should render children passed to it', () => {
    const title = 'Primary Button';
    const component = shallow(<Button title={title} />);
    console.log(component.find(Text).first().props().children);
    expect(component.find(Text).first().props().children).toBe(title);
  });
});
