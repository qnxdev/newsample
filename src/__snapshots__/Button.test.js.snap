// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without issues 1`] = `
<button
  className="Button"
/>
`;
